src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -30,7 +30,32 @@ internal JsonTypeInfo GetOrAddJsonTypeInfo(Type type)
                 InitializeCachingContext();
             }
 
-            return _cachingContext.GetOrAddJsonTypeInfo(type);
+            return GetOrAddJsonTypeInfoEx(type, configured: true, doNotThrow: false)!;
```

Comment:
Nit: I've not seen the "Ex" suffix before. What does it signify?

Reply:
I guess the old WinAPIs habit: ex->extended. I added that so we don't accidentally use wrong overload. I can join them together if you prefer

Reply:
Actually seems I over-complicated 😄

Reply:
I've removed that method in the latest iteration